pkg/virt-handler/virtwrap/api/converter.go

```diff
@@ -48,6 +49,51 @@ func Convert_v1_Disk_To_api_Disk(diskDevice *v1.Disk, disk *Disk) error {
 	return nil
 }
 
+func diskBusFromMachine(bus, machine string) string {
+	if bus != "" {
+		// use what it was explicitely selected
+		return bus
+	}
+	// catches: "q35", "pc-q35-*"
+	// see /path/to/qemu-kvm -machine help
+	if strings.HasPrefix(machine, "pc-q35") || strings.HasPrefix(machine, "q35") {
+		return "sata"
+	}
+	// safe fallback for x86_64, but very slow
+	return "ide"
+}
+
+func diskDeviceFromBusAndIndex(deviceName, bus string, index int) string {
+	if deviceName != "" {
+		return deviceName
+	}
+	prefix := ""
+	switch bus {
+	case "virtio":
+		prefix = "vd"
+	case "sata", "scsi":
+		prefix = "sd"
+	case "ide":
+		prefix = "hd"
+	default:
+		// we should never get there
+		return ""
+	}
+	return formatDeviceName(prefix, index)
+}
+
+// port of http://elixir.free-electrons.com/linux/v4.15/source/drivers/scsi/sd.c#L3211
+func formatDeviceName(prefix string, index int) string {
+	base := int('z' - 'a' + 1)
+	name := prefix
+
+	for index >= 0 {
```

Comment:
This is incorrect. The letters in the output name are backwards.

the 27th disk should be `sdab` not `sdba`.
the 28th disk should be `sdac` not `sdca`...

In the algorithm referenced in the comment, there was a memcpy at the end to reverse the output.

Reply:
Right, good catch! fixed